pandas/io/parsers.py

```diff
@@ -1772,7 +1773,21 @@ def _handle_usecols(self, columns, usecols_key):
                 col_indices = []
                 for u in self.usecols:
                     if isinstance(u, string_types):
-                        col_indices.append(usecols_key.index(u))
+                        if usecols_key.index(u) not in col_indices:
+                            col_indices.append(usecols_key.index(u))
+                        else: # found duplicates in usecols_key.
+                            # get indices of all instances of u in usecols_key. GH11823
+                            u_index = Index(usecols_key).get_loc(u)
+                            # convert the slice or  mask array to index
```

Comment:
what I mean is you don't need a loop and it should be much much simpler, something like
```
(Pdb) p usecols_key 
['a', 'b', 'a']
(Pdb) !res = Index(self.usecols).get_indexer(usecols_key)
(Pdb) p res
array([ 0, -1,  0])
(Pdb) p pd.unique(res)
array([ 0, -1])
```
you can simply ignore the ``-1`` (no column found)